caikit/runtime/service_factory.py

```diff
@@ -42,6 +42,10 @@
     TrainingStatusResponse,
 )
 from caikit.runtime import service_generation
+from caikit.runtime.service_generation.output_target import (
+    PluginFactory as OutputTargetPluginFactory,
```

Comment:
I'd probably be in favor of calling this `OutputTargetPluginFactory` in the definition (and maybe fixing the corresponding one in `DataStreamSource` too)